### DISCUSSION TITLE
How use the feature "fill missing/bad sectors with -=[BAD SECTOR]=- pattern"

### DISCUSSION THREAD

**Discussion by AngelDevIndie** [keirf/greaseweazle#264] [Q&A] [answered]
Anyone how use this feature. It's possible for DOS Disks?

Thanks.

**Comment by keirf** [comment, marked as answer]
If you partially read a disk or disk image, or there are errors in the disk, then those sectors in the output will be filled with this new pattern. This applies to all formats understood by greaseweazle (IBM 720k, 1.44M, AmigaDOS etc etc)

**Reply by GeoKM** [reply]
Hi keirf, I can see the benefits of this in certain situations, but is there an option to switch this behavior off? 

ie. for 1541 often we want to keep the bad sectors etc in their true representation within the flux (scp) image.

In the future you may want to know that a file has been corrupted, and if the blocks now have good CRC's etc they might be slipped into a file as good data. Or obviously, as you would know, many C64 protections (especially the simple ones) rely on finding a specific block error in a certain location.

I've just started imaging all of my C128 & C64 disks as flux SCP's, granted 99% of my disks are not protected, but I do want to know that bad blocks are bad blocks.

Prehaps, keep the bad sector flux data in the SCP by default, and provide an option to replace it with [BAD SECTOR] on convert or write?

8-)

Thanks for a great device BTW 8-)
